trio/__init__.py

```diff
@@ -13,61 +13,81 @@
 # This file pulls together the friendly public API, by re-exporting the more
 # innocuous bits of the _core API + the higher-level tools from trio/*.py.
 
-from ._version import __version__
+__all__ = [
+    "TrioInternalError",
+    "RunFinishedError",
+    "WouldBlock",
+    "Cancelled",
+    "ResourceBusyError",
+    "ClosedResourceError",
+    "MultiError",
+    "run",
+    "open_nursery",
+    "open_cancel_scope",
+    "current_effective_deadline",
+    "TASK_STATUS_IGNORED",
+    "current_time",
+    "TaskLocal",
+]
 
-__all__ = []
+from ._version import __version__
 
-from ._toplevel_core_reexports import *
-__all__ += _toplevel_core_reexports.__all__
+from ._core import (
+    TrioInternalError, RunFinishedError, WouldBlock, Cancelled,
+    ResourceBusyError, ClosedResourceError, MultiError, run, open_nursery,
+    open_cancel_scope, current_effective_deadline, TASK_STATUS_IGNORED,
+    current_time, TaskLocal, cancel_shielded_checkpoint
+)
 
-from ._timeouts import *
-__all__ += _timeouts.__all__
+from ._timeouts import (
+    move_on_at, move_on_after, sleep_forever, sleep_until, sleep, fail_at,
+    fail_after, TooSlowError
+)
 
-from ._sync import *
-__all__ += _sync.__all__
+from ._sync import (
+    Event, CapacityLimiter, Semaphore, Lock, StrictFIFOLock, Condition, Queue
+)
 
-from ._threads import *
-__all__ += _threads.__all__
+from ._threads import (
+    run_sync_in_worker_thread, current_default_worker_thread_limiter,
+    BlockingTrioPortal
+)
 
-from ._highlevel_generic import *
-__all__ += _highlevel_generic.__all__
+from ._highlevel_generic import (
+    aclose_forcefully, BrokenStreamError, StapledStream
+)
 
-from ._signals import *
-__all__ += _signals.__all__
+from ._signals import catch_signals
```

Comment:
Since #619 just got merged, you'll want to update this to also import `open_signal_receiver`.